flake.nix

```diff
@@ -0,0 +1,80 @@
+{
```

Comment:
Instead of this, how about something like https://gitlab.com/exarkun/chk.hs/-/blob/master/flake.nix ?

I really want to avoid having 10 slightly different flakes for Haskell projects.  To the extent that hs-flake-utils can't do something we want, we should improve hs-flake-utils (to provide it directly or parameterize the missing pieces).